### PR TITLE
Decompile unk_02031AF0.s

### PR DESCRIPTION
**Pull request by adrienntindall**
Something to do with unk save data 40 and 29, not much there, and I chose this smaller file since this was my first day doing any sort of decompiling. Not sure if I did it right, but it passes the sha1 test and compiles/runs properly so I guess it's prolly fine.
I'll try and get better and contribute more in the future when I can =)

### CODE REVIEW

File: include/unk_02031AF0.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#ifndef POKEHEARTGOLD_UNK_02031AF0_H
+#define POKEHEARTGOLD_UNK_02031AF0_H
+
+#include "save.h"
+
+u16 sub_02031AF0(); //02031AF0
```
Comment (tgsm) [inline include/unk_02031AF0.h:6]: This comment is redundant

Also, any functions that don't take in any parameters should just have `void`:
```suggestion
u32 sub_02031AF0(void);
```

File: include/unk_02031AF0.h [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+void sub_02031AF4(u16* unk);
+u16* sub_02031B00(SAVEDATA* savedata);
+
+#endif //POKEHEARTGOLD_UNK_02031AF0_H
```
Comment (tgsm) [inline include/unk_02031AF0.h:10]: Files need a newline at the end

File: src/unk_02031AF0.c [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+u16 sub_02031AF0() { 
+    return SAVE_UNK_40;
+}
+void sub_02031AF4(u16* unk) {
+    MI_CpuFill8(unk, 0, SAVE_UNK_40);
+}
+
+u16* sub_02031B00(SAVEDATA* savedata) {
+    return SavArray_get(savedata, SAVE_UNK_29);
+}
```
Comment (tgsm) [inline src/unk_02031AF0.c:13]: Ditto

File: src/unk_02031AF0.c [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#include "unk_02031AF0.h"
+#include "save.h"
+
+u16 sub_02031AF0() { 
+    return SAVE_UNK_40;
+}
+void sub_02031AF4(u16* unk) {
```
Comment (tgsm) [inline src/unk_02031AF0.c:7]: Need a newline here separating the two functions

**Review by tgsm — CHANGES_REQUESTED** [review]
A good first try, but there are some changes that need to be made.

* So this file contains a size function and an initialize function for save struct 29. Every save struct has these. The size function (`sub_02031AF0`) function simply returns the size of the struct. First you'll have to define the struct in the header. Something basic like this is fine for now:
```c
typedef struct {
    u8 unk[0x24];
} UnkSaveStruct29;
```
Then all you gotta do is return the size of that:
```c
u32 sub_02031AF0(void) {
    return sizeof(UnkSaveStruct29);
}
```

* The initialize function (`sub_02031AF4`) takes in a pointer to the save struct that will be initialized, not a u16*. We also pass the size of the struct, not the ID of another save struct.
```c
void sub_02031AF4(UnkSavStruct29* unk) {
    MI_CpuFill8(unk, 0, sizeof(sub_02031AF0));
}
```

* Lastly, for the getter function (`sub_02031B00`) all that needs to be changed is the return type to return our save struct.

I've also attached some comments just pointing out some other minor nits that should be addressed.

**Comment by adrienntindall** [comment]
Added all the requested changes, thanks so much for the feedback! I definitely helped me understand what I'm doing a bit better.

File: include/unk_02031AF0.h [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#ifndef POKEHEARTGOLD_UNK_02031AF0_H
+#define POKEHEARTGOLD_UNK_02031AF0_H
+
+#include "save.h"
+
+typedef struct {
+	u8 unk[0x28]; 
+} UnkSaveStruct29;
+
+u16 sub_02031AF0(void);
```
Comment (tgsm) [inline include/unk_02031AF0.h:10]:
```suggestion
u32 sub_02031AF0(void);
```

File: include/unk_02031AF0.h [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+#define POKEHEARTGOLD_UNK_02031AF0_H
+
+#include "save.h"
+
+typedef struct {
+	u8 unk[0x28]; 
+} UnkSaveStruct29;
+
+u16 sub_02031AF0(void); 
+void sub_02031AF4(u16* unk);
```
Comment (tgsm) [inline include/unk_02031AF0.h:11]:
```suggestion
void sub_02031AF4(UnkSaveStruct29* unk);
```

File: src/unk_02031AF0.c [hunk 7]
```diff
@@ -0,0 +1,14 @@
+#include "unk_02031AF0.h"
+#include "save.h"
+
+u16 sub_02031AF0() {
```
Comment (tgsm) [inline src/unk_02031AF0.c:4]:
```suggestion
u32 sub_02031AF0(void) {
```

File: src/unk_02031AF0.c [hunk 8]
```diff
@@ -0,0 +1,14 @@
+#include "unk_02031AF0.h"
+#include "save.h"
+
+u16 sub_02031AF0() { 
+    return sizeof(UnkSaveStruct29);
+}
+
+void sub_02031AF4(u16* unk) {
```
Comment (tgsm) [inline src/unk_02031AF0.c:8]:
```suggestion
void sub_02031AF4(UnkSaveStruct29* unk) {
```

**Review by tgsm — CHANGES_REQUESTED** [review]
Good job catching the typo, sorry about that!

**Review by tgsm — APPROVED** [review]

**Pull request merged**